lib/msal-react/src/hooks/useAccount.ts

```diff
@@ -51,25 +53,42 @@ export function useAccount(
         }
     });
 
+    const updateAccount = useCallback(() => {
+        const nextAccount = getAccount(instance, accountIdentifiers);
+        setAccount((currentAccount: AccountInfo | null) => {
+            if (
+                !AccountEntityUtils.accountInfoIsEqual(
+                    currentAccount,
+                    nextAccount,
+                    true
+                )
+            ) {
+                logger.info("useAccount - Updating account");
+                return nextAccount;
+            }
+            return currentAccount;
+        });
+    }, [accountIdentifiers, instance, logger]);
+
     useEffect(() => {
         if (inProgress !== InteractionStatus.Startup) {
-            setAccount((currentAccount: AccountInfo | null) => {
-                const nextAccount = getAccount(instance, accountIdentifiers);
-                if (
-                    !AccountEntityUtils.accountInfoIsEqual(
-                        currentAccount,
-                        nextAccount,
-                        true
-                    )
-                ) {
-                    logger.info("useAccount - Updating account");
-                    return nextAccount;
+            updateAccount();
+        }
+
+        const callbackId = instance.addEventCallback(
+            (message: EventMessage) => {
+                if (message.eventType === EventType.ACTIVE_ACCOUNT_CHANGED) {
```

Comment:
Need to account for ssoSilent now too since we aren't setting the inProgress state for it anymore
```suggestion
                if (message.eventType === EventType.ACTIVE_ACCOUNT_CHANGED || (message.eventType === EventType.LOGIN_SUCCESS && message.interactionType === InteractionType.Silent)) {
```